gpkit/keydict.py

```diff
@@ -131,6 +140,15 @@ def __getitem__(self, key):
     def __setitem__(self, key, value):
         "Overloads __setitem__ and []= to work with all keys"
         key, idx = self.parse_and_index(key)
+        if hasattr(value, "exp") and not value.exp:
+            value = value.value  # substitute constant monomials
```

Comment:
this seems surprisingly similar to the first two lines of `clean_value`? Maybe no way around that